dspace-server-webapp/src/main/java/org/dspace/app/rest/model/hateoas/HALResource.java

```diff
@@ -49,6 +51,20 @@ public void setPageHeader(EmbeddedPageHeader page) {
     public EntityModel<T> add(Link link) {
         if (!hasLink(link.getRel())) {
             return super.add(link);
+        } else {
+            String name = link.getName();
+            if (StringUtils.isNotBlank(name)) {
+                List<Link> list = this.getLinks(link.getRel());
+                boolean doesNameExist = false;
+                for (Link existingLink : list) {
```

Comment:
While this entire `for` loop should work, just a note that this can be simplified in Java 8+ like this:
```
// If a link of this name doesn't already exist in the list, add it
if (!list.stream().anyMatch((l -> l.getName().equalsIgnoreCase(name)))) {
    super.add(link);
}
```

If you prefer your manner, that's fine. It'll work.  Just noting an opportunity for code simplification here.

Reply:
@tdonohue Thanks for this point of feedback, we altered the code in this manner (& also merged the latest master into this branch).